plan/analyze_test.go

```diff
@@ -97,9 +100,18 @@ func (s *testAnalyzeSuite) TestAnalyze(c *C) {
 		is := sessionctx.GetDomain(ctx).InfoSchema()
 		err = plan.ResolveName(stmt, is, ctx)
 		c.Assert(err, IsNil)
-		err = plan.InferType(ctx.GetSessionVars().StmtCtx, stmt)
+		err = expression.InferType(ctx.GetSessionVars().StmtCtx, stmt)
 		c.Assert(err, IsNil)
 		p, err := plan.Optimize(ctx, stmt, is)
 		c.Assert(plan.ToString(p), Equals, tt.best, Commentf("for %s", tt.sql))
 	}
 }
+
+func newStoreWithBootstrap() (kv.Storage, error) {
```

Comment:
This function was defined in typeinferer.go and used by some test-function in package plan,
but I move typeinferer.go from package plan to package expression in this commit,
so I port it here.